### PR TITLE
Fix shape bugs

### PR DESCRIPTION
**Pull request by Vimalakirti**
Fixed a lot of shape issues across the repo. (mainly due to padding and special cases)

### CODE REVIEW

File: src/onnx.rs [hunk 1]
```diff
@@ -75,6 +75,9 @@ fn parse_onnx_constants<'a>(
       DatumType::F32 => {
         let tensor = tensor.into_array::<f32>().unwrap();
         Ok(tensor.map(|x| {
+          if *x < 1e-10 && *x > 0.0 {
```
Comment (ddkang) [inline src/onnx.rs:79]: What'as this?

File: src/util/shape.rs [hunk 2]
```diff
@@ -3,6 +3,7 @@
  * The functions are used for shape-related operations, such as
  * slicing and padding arrays.
  */
+#![allow(dead_code)]
```
Comment (ddkang) [inline src/util/shape.rs]: Why this?

File: src/onnx.rs (hunk 1 above)
Comment (Vimalakirti) [reply src/onnx.rs:79]: When the scale factor is not big enough, an epsilon constant will be 0, which might cause DivisionByZero error. This is a workaround solution for Bert and GPTj.

File: src/util/shape.rs (hunk 2 above)
Comment (Vimalakirti) [reply src/util/shape.rs]: We currently do not use the function `slice_nd_array`,  but we might need to use it later. `#![allow(dead_code)]` is a tag which helps ignore the warning that a function is unused.

File: src/onnx.rs (hunk 1 above)
Comment (ddkang) [reply src/onnx.rs:79]: This minimally needs a comment as to why this is 1

File: src/util/shape.rs (hunk 2 above)
Comment (ddkang) [reply src/util/shape.rs]: Can you put it above the relevant function?

**Review by liliatangxy — DISMISSED** [review]

File: src/onnx.rs (hunk 1 above)
Comment (Vimalakirti) [reply src/onnx.rs:79]: added comments

File: src/util/shape.rs (hunk 2 above)
Comment (Vimalakirti) [reply src/util/shape.rs]: moved the tag above the function

**Review by ddkang — APPROVED** [review]

**Pull request merged**